test/test-fs-event.c

```diff
@@ -932,6 +932,92 @@ TEST_IMPL(fs_event_getpath) {
   return 0;
 }
 
+static uv_timer_t *end_common_prefix_timer;
+static void fs_event_cb_common_prefix(uv_fs_event_t* handle, const char* filename,
+  int events, int status) {
+  char buf[1024];
+  size_t buf_size;
+
+  ASSERT(status == 0);
+  buf_size = sizeof(buf) - 1;
+  uv_fs_event_getpath(handle, buf, &buf_size);
+  if (buf_size >= sizeof(buf))
+    buf_size = sizeof(buf) - 1;
+
+  printf("fs_event_cb: %s/%s\n", buf, filename);
+  ++fs_event_cb_called;
+  uv_timer_again(end_common_prefix_timer);
+}
+
+static void end_common_prefix_test_cb(uv_timer_t *handle) {
+  uv_stop(uv_default_loop());
+}
+
+TEST_IMPL(fs_event_common_prefix) {
+#if defined(NO_FS_EVENTS)
+  RETURN_SKIP(NO_FS_EVENTS);
+#endif
+  uv_loop_t* loop;
+  int r;
+  uv_fs_event_t fs_event1;
+  uv_fs_event_t fs_event2;
+  uv_timer_t timer;
+
+  /* Setup */
+  create_dir("watch_dir");
+  create_dir("watch_dir2");
+  create_file("watch_dir/file1");
+  create_file("watch_dir2/file1");
+
+  loop = uv_default_loop();
+  r = uv_timer_init(loop, &timer);
+  ASSERT(r == 0);
+  end_common_prefix_timer = &timer;
+
+  r = uv_fs_event_init(loop, &fs_event1);
+  ASSERT(r == 0);
+  r = uv_fs_event_start(&fs_event1,
+                        fs_event_cb_common_prefix,
+                        "watch_dir",
+                        0);
+  ASSERT(r == 0);
+  r = uv_fs_event_init(loop, &fs_event2);
+  ASSERT(r == 0);
+  r = uv_fs_event_start(&fs_event2,
+                        fs_event_cb_common_prefix,
+                        "watch_dir2",
+                        0);
+  ASSERT(r == 0);
+
+  touch_file("watch_dir/file1");
+  touch_file("watch_dir2/file1");
```

Comment:
By looking at the test again I think this is not correct.

I think what you want is to make the file modifications in the `timer` callback otherwise the events can be missed. See what we do in some other fs_tests: https://github.com/libuv/libuv/blob/v1.x/test/test-fs-event.c#L396-L435.
Then you can start a timer on the fs_event callback to make sure only the events you want are indeed received.

Also, as you're watching specifically directories I think some platforms won't report about modifications in files inside that directory (I would like some confirmation on this as I've observed this behaviour on `FreeBSD`). Adding / removing files should work fine.